xapian-letor/api/letor_internal.cc

```
@@ -262,15 +262,62 @@ Letor::Internal::prepare_training_file(const string & queryfile, const string &
     fm.set_database(letor_db);
     fm.load_relevance(qrel_file);
 
-    list<Xapian::RankList> l;
+    vector<Xapian::RankList> l;
 
     string str1;
     ifstream myfile1;
     myfile1.open(queryfile.c_str(), ios::in);
 
+    if(!myfile1.good()){
+        cout << "No Query file found"<<endl;
```

Should raise an exception.